### PR TITLE
[Refactor] StockPile.Save now uses helper functions

### PR DESCRIPTION
**Pull request by scowalt**
`StockPile.Save` is a relatively large and complex function that performs many different actions.

I've done my best to componentize this function into helper functions in a logical way and with logical groupings. This should make this function much more maintainable and easy to expand upon in the future.

### CODE REVIEW

File: Source/Libraries/CorruptCore/Stockpile/SaveException.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+namespace RTCV.CorruptCore.StockpileNS
```
Comment (NarryG) [inline Source/Libraries/CorruptCore/Stockpile/SaveException.cs:1]: Not a huge fan of this namespace. Maybe we could pull exceptions into a common namespace?

Comment (scowalt) [reply Source/Libraries/CorruptCore/Stockpile/SaveException.cs:1]: Agreed, I don't love `RTCV.CorruptCore.StockpileNS`. I could create `RTCV.CorruptCore.Exceptions` (with a `CorruptCore\Exceptions` folder) and put it there.

This is the only custom exception that I could find in `CorruptCore`. However, I have a feeling there may be more on the way (like a `LoadException`)

Comment (scowalt) [reply Source/Libraries/CorruptCore/Stockpile/SaveException.cs]: Done in the latest commit

Comment (NarryG) [reply Source/Libraries/CorruptCore/Stockpile/SaveException.cs]: Sounds good. I do agree that more custom exceptions will make sense which is why I brought this up

**Pull request merged**